### ISSUE TITLE
libmpv: playback-time not updated on a frame basis when observing with mpv_observe_property

### ISSUE THREAD

**Issue by gunnarmarten** [mpv-player/mpv#4195]
### mpv version and platform

MacOS Sierra 10.12.3
mpv: HEAD-9740383, 0.24, 0.23, 0.22
ffmpeg: 3.2.4

### Reproduction steps
I'm using libmpv with opengl-cb in my opengl application. Like in a video player I want to know the current position or current frame in a playing video.
I used mpv_observe_property for that. With recent mpv versions this is not working the property is only updated once in a while (also the properties 'percent-pos' and 'time-pos')

If I remember it correctly it worked with older mpv versions.

To reproduce:
Use the sdl example from https://github.com/mpv-player/mpv-examples.git

Observe the property:
```
mpv_observe_property(mpv, 0, "playback-time", MPV_FORMAT_DOUBLE);
```

and wait for property changed events in the event loop, like this:

```
if (mp_event->event_id == MPV_EVENT_PROPERTY_CHANGE) {
    mpv_event_property *prop = (mpv_event_property *)mp_event->data;
    printf("MPV_EVENT_PROPERTY_CHANGE %s\n", prop->name);
    if (strcmp(prop->name, "playback-time") == 0 && prop->format == MPV_FORMAT_DOUBLE) {
        double pos = *(double*)prop->data;
        printf("playback-time: %f\n", pos);
    }
}

```
### Expected behavior

Should update the playback-time every frame.

### Actual behavior

Updates the property only sometimes. See attached log.
A test with mpv 0.18 showed it updates the property every second frame (see attached log)

### Log file

with mpv master: http://sprunge.us/XgQg
with mpv 0.18: http://sprunge.us/gXUT

### Sample files

Any video.

**Comment by wm4** [comment]
Can't reproduce with the sdl example and Linux.

**Comment by gunnarmarten** [comment]
yeah, just tested the sdl example on Ubuntu 14.04 with mpv 0.24 and it works, every frame I get an update.
So an OSX related issue?
I can try getting a windows in my hand and test the sdl example there.

**Comment by gunnarmarten** [comment]
ok did a test on Windows 7 64 bit with mpv 0.24
It updates `playback-time` more frequently but not every frame, see logfile http://sprunge.us/KVTG

**Comment by gunnarmarten** [comment]
Hey, is there any information I can provide to resolve this?

Is observing a property a bad approach for frame-based updates, should I get the property `playback-time` every time the new video frame is ready?

**Comment by gunnarmarten** [comment]
I did a git bisect to find the first 'bad' revision that changed the behavior of observing `playback-time` on OSX.
The bisect showed this commit: 

> b0ef3dd4fb49b8fe111b00ca44c30b9890be0282 is the first bad commit
> commit b0ef3dd4fb49b8fe111b00ca44c30b9890be0282
> Author: wm4 <wm4@nowhere>
> Date:   Sun Oct 30 12:16:24 2016 +0100
> 
>     vo_opengl_cb: fix inverted condition
> 
>     Commit e6291697 got this wrong.

With this commit the `playback-time` only updates sporadically. Before it updates regularly.
See the logfiles from the first comment of this issue:
before: http://sprunge.us/gXUT
after : http://sprunge.us/XgQg

I hope this helps.

**Comment by gunnarmarten** [comment]
I revisited my old issue here, it is still an issue with the current master.

So I did some debugging based on the fact that a git bisect from 2 years ago (see last comment) found a commit that triggers the bug.
What I found is, that the calling order of the property update method in the client API changed on MacOS with commit e6291697 and the bugfix commit b0ef3dd.

For the debugging I used again the sdl example from https://github.com/mpv-player/mpv-examples.git

Execution order BEFORE b0ef3dd:
```
mark_property_changed playback-time
gen_property_change_event property: playback-time changed: 1
gen_property_change_event property: playback-time changed: 0
draw frame
update_prop playback-time
gen_property_change_event property: playback-time changed: 1
MPV_EVENT_PROPERTY_CHANGE playback-time
playback-time: 3.666667
gen_property_change_event property: playback-time changed: 0
mark_property_changed playback-time
gen_property_change_event property: playback-time changed: 1
gen_property_change_event property: playback-time changed: 0
draw frame
update_prop playback-time
gen_property_change_event property: playback-time changed: 1
MPV_EVENT_PROPERTY_CHANGE playback-time
playback-time: 3.700000
```

Execution order AFTER b0ef3dd:

```
mark_property_changed playback-time
gen_property_change_event property: playback-time changed: 1
gen_property_change_event property: playback-time changed: 0
update_prop playback-time
draw frame
mark_property_changed playback-time
gen_property_change_event property: playback-time changed: 1
gen_property_change_event property: playback-time changed: 0
update_prop playback-time
draw frame
mark_property_changed playback-time
gen_property_change_event property: playback-time changed: 1
gen_property_change_event property: playback-time changed: 0
draw frame
update_prop playback-time
```

So after every successful update_prop() where the prop->changed flag is set, mark_property_changed() is called and also sets prop->changed. But additionally the prop->need_new_value is set. This causes that the next call to gen_property_change_event() requests another property update and omits the outstanding MPV_EVENT_PROPERTY_CHANGE event. As a result the property change event is never called for frame-based properties like `playback-time`

Both mark_property_changed() and update_prop() use the same flag (prop->changed) to signal different things. The update_prop() should signal a successful update. I fixed that in PR #6642

Execution order after fix from PR #6642:

```
mark_property_changed playback-time
gen_property_change_event property: playback-time changed: 1
MPV_EVENT_PROPERTY_CHANGE playback-time
playback-time: 2.700000
gen_property_change_event property: playback-time changed: 0
gen_property_change_event property: playback-time changed: 0
draw frame
update_prop playback-time
mark_property_changed playback-time
gen_property_change_event property: playback-time changed: 1
MPV_EVENT_PROPERTY_CHANGE playback-time
playback-time: 2.733333
gen_property_change_event property: playback-time changed: 0
gen_property_change_event property: playback-time changed: 0
draw frame
update_prop playback-time
mark_property_changed playback-time
gen_property_change_event property: playback-time changed: 1
MPV_EVENT_PROPERTY_CHANGE playback-time
playback-time: 2.766667

```
